subsys/net/ip/net_if.c

```diff
@@ -555,6 +561,90 @@ static void dad_timeout(struct k_work *work)
 		 */
 		net_ipv6_nbr_rm(iface, &ifaddr->address.in6_addr);
 	}
+
+	ifaddr->dad.timer_timeout = 0;
+
+	sys_slist_find_and_remove(&active_dad_timers, &ifaddr->dad.node);
+}
+
+static bool dad_check_timeout(s64_t start, u32_t time, s64_t timeout)
+{
+	start += time;
+	start = abs(start);
+
+	if (start > timeout) {
+		return false;
+	}
+
+	return true;
+}
+
+static bool dad_timedout(struct net_if_addr *ifaddr, s64_t timeout)
+{
+	return dad_check_timeout(ifaddr->dad.timer_start,
+				 ifaddr->dad.timer_timeout,
+				 timeout);
+}
+
+static u32_t dad_manage_timeouts(struct net_if_addr *ifaddr, s64_t timeout)
+{
+	s32_t next_timeout;
+
+	if (dad_timedout(ifaddr, timeout)) {
+		dad_expired(ifaddr);
+		return UINT32_MAX;
+	}
+
+	next_timeout = timeout - (ifaddr->dad.timer_start +
```

Comment:
Very confusing arithmetics, mixing up 32 and 64 bit numbers, signed and unsigned.

Reply:
We just need the diff of two s64_t values, and that diff will fit s32_t

Reply:
No, there're no s64_t values! There're only u64_t values disguised as s64_t, as explained in the previous comment. Please avoid using hosed k_uptime_get(), instead use sanely-defined k_uptime_get_32() (wraparound period 49.7 days, more than enough for networking needs), and leave all these problems behind.